processing/src/main/java/io/druid/granularity/QueryGranularity.java

```diff
@@ -86,4 +89,74 @@ else if(o instanceof Number)
     }
     throw new IAE("Cannot convert [%s] to QueryGranularity", o.getClass());
   }
+
+  // best effort comparing
+  public static Integer compare(QueryGranularity g1, QueryGranularity g2) {
```

Comment:
it would be better to have 
```
public abstract Integer compare(QueryGranularity granularity);
```
and each implementation implementing it, rather than having it all in one place for all concrete types.